torch/_meta_registrations.py

```diff
@@ -5763,6 +5763,8 @@ def argmax_argmin_meta(self, dim=None, keepdim=False):
 
 @register_meta(aten.scalar_tensor.default)
 def scalar_tensor(s, dtype=None, layout=None, device=None, pin_memory=None):
```

Comment:
`SymFloat` represents a python float that are tracking symbolically. It subtypes float and should have all the same properties. 

```
>>> torch.scalar_tensor(3.5).dtype
torch.float32
```

This is the not the right place to put it. We should change the constructor of wherever we are instantiating the symfloat with float32 to pass in float64 explicitly, which also makes clear at the callsite how we are handling it, instead of deep in the dtype propagation analysis

Reply:
As discussed offline, here is where the symfloat is converted to a faketensor. So, it should be correct to set the dtype here.

Reply:
Oh, because we have handled the dtype in the upper level's `as_tensor`, the argument dtype has been successfully passed here.